apps/meteor/tests/e2e/admin-users.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { faker } from '@faker-js/faker';
+
+import { Users } from './fixtures/userStates';
+import { Admin } from './page-objects';
+import { expect, test } from './utils/test';
+
+test.use({ storageState: Users.admin.state });
+
+test(
+	'New user shows in correct tabs when deactivated',
+	{
+		tag: '@admin',
+		annotation: {
+			type: 'issue',
+			description: 'https://rocketchat.atlassian.net/browse/SUP-775',
+		},
+	},
+	async ({ page }) => {
+		const poAdmin = new Admin(page);
+		await page.goto('/admin/users');
+
+		const { username } = await createUser(poAdmin);
+
+		await page.getByPlaceholder('Search Users').fill(username);
+
+		await test.step('is visible in the All tab', async () => {
+			await page.getByRole('tab', { name: 'All' }).click();
+			await expect(page.getByRole('link', { name: username })).toBeVisible();
+		});
+
+		await test.step('is visible in the Pending tab', async () => {
+			await page.getByRole('tab', { name: 'Pending' }).click();
+			await expect(page.getByRole('link', { name: username })).toBeVisible();
+		});
+
+		await test.step('is not visible in the Active tab', async () => {
+			await page.getByRole('tab', { name: 'Active' }).click();
+			await expect(page.getByRole('link', { name: username })).not.toBeVisible();
+		});
+
+		await test.step('is not visible in the Deactivated tab', async () => {
+			await page.getByRole('tab', { name: 'Deactivated' }).click();
+			await expect(page.getByRole('link', { name: username })).not.toBeVisible();
+		});
+
+		await test.step('moves from Pending to Deactivated tab', async () => {
+			await page.getByRole('tab', { name: 'Pending' }).click();
+			await page.getByRole('button', { name: 'More actions' }).click();
+			await page.getByRole('menuitem', { name: 'Deactivate' }).click();
+			await expect(page.getByRole('link', { name: username })).not.toBeVisible();
+
+			await page.getByRole('tab', { name: 'Deactivated' }).click();
+			await expect(page.getByRole('link', { name: username })).toBeVisible();
+		});
+
+		await test.step('moves from Deactivated to Pending tab', async () => {
+			await page.getByRole('tab', { name: 'Deactivated' }).click();
+			await page.getByRole('button', { name: 'More actions' }).click();
+			await page.getByRole('menuitem', { name: 'Activate' }).click();
+			await expect(page.getByRole('link', { name: username })).not.toBeVisible();
+
+			await page.getByRole('tab', { name: 'Pending' }).click();
+			await expect(page.getByRole('link', { name: username })).toBeVisible();
+		});
+	},
+);
+
+async function createUser(poAdmin: Admin) {
```

Comment:
Maybe you can use rest API to create a new user to test this, or if we want it this way, we can move this utility function in a file of its own as a better code practice.

Reply:
I'm leaning towards not using the rest api directly, so I'll follow your advice about moving the utility to its own file.

However, if you have a case for using the rest API directly, let me know.

Reply:
I'll try that first then, thanks!